spec/lib/services/redfish-obm-service-spec.js

```diff
@@ -0,0 +1,122 @@
+                'reset_type@Redfish.AllowableValues': [
+                    'On',
+                    'ForceOn',
+                    'GracefulShutdown',
+                    'ForceOff',
+                    'GracefulRestart',
+                    'ForceRestart',
+                    'Nmi',
+                    'PushPowerButton'
+                ]}
+            }
+        ];
+        
+    before(function() {
+        redfishApi = Promise.promisifyAll(new redfish.RedfishvApi());
+        sandbox.stub(redfishApi);
+
+        redfishApi.listResetTypesAsync.resolves(testData);
+        redfishApi.doResetAsync.resolves([
+            null,
+            { body: {'@odata.id': 'abc123'} }
+        ]);
+    });
+    
+    after(function() {
+        sandbox.restore();
+    });
+    
+    describe('redfish service force false', function() {     
+        base.before('before', servicePath, function(self) {
+            self.serviceOptions = testOptions;
+            self.Service = helper.injector.get('redfish-obm-service');
+            sandbox.stub(self.Service.prototype, '_initClient');
+            self.Service.prototype._initClient.returns(redfishApi)
+                .returns(redfishApi);
+        });
+        
+        base.runInterfaceTestCases([
+            'powerOn',
+            'powerOff',
+            'reboot',
+            'NMI',
+            'powerButton',
+            'powerStatus'
+        ]);
+    });
+    
+    describe('redfish service force true', function() {  
+        base.before('before', servicePath, function(self) {
+            testOptions.params.force = true;
+            self.serviceOptions = testOptions;
+            self.Service = helper.injector.get('redfish-obm-service');
+            sandbox.stub(self.Service.prototype, '_initClient')
+                .returns(redfishApi);
+        });
+        
+        base.runInterfaceTestCases([
+            'powerOn',
+            'powerOff',
+            'reboot'
+        ]);
+    });
+    
+    describe('redfish service initialize client', function() {  
+        var redfishService, baseObm;
+        base.before('before', servicePath, function(self) {
+            redfishService = helper.injector.get('redfish-obm-service');
+            baseObm = helper.injector.get('OBM.base');
+        });  
+        
+        it ('should with credentials', function() {
+            redfishService.create(testOptions);
+            expect(redfishService).to.be.ok;
+            expect(baseObm).to.be.ok;
+            expect(baseObm.create).to.have.been.calledWith(redfishService);
+        });
+    }); 
+    
+    describe('redfish service initialize client', function() {  
+        var redfishService, baseObm;
+        base.before('before', servicePath, function(self) {
+            redfishService = helper.injector.get('redfish-obm-service');
+            baseObm = helper.injector.get('OBM.base');
+        });  
+        
+        it ('should with no credentials', function() {
```

Comment:
It is, but that's not what will be printed:

"redfish service initialize client should with no credentials" is what will show up.